app/questionnaire/router.py

```diff
@@ -124,15 +124,12 @@ def get_next_location_url(
         if self._progress_store.is_section_complete(
             location.section_id, location.list_item_id
         ):
-            if return_to == "section-summary":
-                return self._get_section_url(location)
-
-            if return_to == "final-summary" and self.is_questionnaire_complete:
-                return url_for("questionnaire.submit_questionnaire")
-
             if is_last_block_in_section:
                 return self._get_next_location_url_for_last_block_in_section(location)
 
+            if return_to:
+                return self._get_return_to_location_url(location, return_to)
```

Comment:
How come the order changes?. Also `_get_return_to_location_url` may not return anything, so this needs to check you have a return to and are able to get a return to url.

eg:
```py
if return_to and (return_to_url := self._get_return_to_location_url(location, return_to)):
	return return_to_url

Reply:
Reverted order